tests/providers/common/io/datasets/test_file.py

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from urllib.parse import urlsplit, urlunsplit
+
+import pytest
+from openlineage.client.run import Dataset as OpenLineageDataset
+
+from airflow.datasets import Dataset
+from airflow.providers.common.io.datasets.file import convert_dataset_to_openlineage, sanitize_uri
+
+
+@pytest.mark.parametrize(
+    ("uri", "expected"),
+    (
+        ("file:///valid/path/", "file:///valid/path/"),
+        ("file://C://dir/file", "file://C://dir/file"),
+    ),
+)
+def test_sanitize_uri_valid(uri, expected):
+    result = sanitize_uri(urlsplit(uri))
+    assert urlunsplit(result) == expected
+
+
+@pytest.mark.parametrize("uri", ("file://",))
+def test_sanitize_uri_invalid(uri):
+    with pytest.raises(ValueError):
+        sanitize_uri(urlsplit(uri))
+
+
+@pytest.mark.parametrize(
+    ("uri", "ol_dataset"),
+    (
+        ("file:///valid/path/", OpenLineageDataset(namespace="file://", name="/valid/path")),
+        ("file:///valid/path", OpenLineageDataset(namespace="file://", name="/valid/path")),
+        ("file://C://dir/file", OpenLineageDataset(namespace="file://C:", name="//dir/file")),
```

Comment:
Should `C:` be a `netloc` part of URL (dataset namespace), or a part of file path (dataset name)?

Reply:
It should be in the path, i think. I improved the test, used another valid uri format.